scripts/batch.py

```
@@ -224,399 +197,98 @@ async def monitor_processes(self) -> None:
+        loop = asyncio.new_event_loop()
+        asyncio.set_event_loop(loop)
 
-                def signal_handler():
-                    self.shutdown_event.set()
+        try:
+            def signal_handler():
+                self.signal_handler(None, None)
 
-                loop.add_signal_handler(signal.SIGINT, signal_handler)
-                loop.add_signal_handler(signal.SIGTERM, signal_handler)
+            loop.add_signal_handler(signal.SIGINT, signal_handler)
```

Ruff (0.8.2)

Undefined name `signal`

(F821)